assignments/hw1/ft_register.go

```diff
@@ -1,27 +1,95 @@
 package hw1
 
 import (
+	"github.com/dati-mipt/distributed-systems/util"
 	"github.com/dati-mipt/distributed-systems/network"
 )
 
 type FaultTolerantRegister struct {
+	rid      int64
+	current  util.TimestampedValue
+	replicas map[int64]network.Link
 }
 
 func NewFaultTolerantRegister(rid int64) *FaultTolerantRegister {
-	return &FaultTolerantRegister{}
+	return &FaultTolerantRegister{
+		rid:      rid,
+		replicas: map[int64]network.Link{},
+	}
 }
 
 func (r *FaultTolerantRegister) Write(value int64) bool {
-	return false
+	cur := r.read()
+
+	cur.Ts.Rid = r.rid
+	cur.Ts.Number++
+	cur.Val = value
+
+	r.current.Store(cur)
+
+	r.write(cur)
+	return true
 }
 
 func (r *FaultTolerantRegister) Read() int64 {
-	return 0
+	cur := r.read()
+	r.write(cur)
+	return cur.Val
+}
+
+func (r *FaultTolerantRegister) write(cur util.TimestampedValue) {
```

Comment:
Actually, write should also contain message sending to at almost quorum recipients.